### DISCUSSION TITLE
How can i add admin middleware for filament?

### DISCUSSION THREAD

**Discussion by tienhan2k** [filamentphp/filament#7797] [Help]
Hello guys, how can i add a middleware for filament? i dont want normal user access to admin pages. Pls help me

**Comment by RickDBCN** [comment]
Hi @tienhan2k,

It is all explained in the documentation:

V3: https://filamentphp.com/docs/3.x/panels/users
V2: https://filamentphp.com/docs/2.x/admin/users

**Reply by tienhan2k** [reply]
thank you a lot sir, its work
